### ISSUE TITLE
Ninja segfault with missing var in description

### ISSUE THREAD

**Issue by rgeary1** [ninja-build/ninja#594]
The following ninja file segfaults :

``` ninja
rule echo
  command=echo $in
  description=$description

build blank: echo
  #description=hi
```

Uncommenting the '#description' build edge override fixes the issue, but I would expect a blank description instead of a segfault.

**Comment by ret2libc** [comment]
This seems to be fixed to me. Trying the testcase in the first comment I get:
`ninja: fatal: cycle in rule variables: description -> description`.

**Comment by nico** [comment]
Yup, independently rediscovered in #902 and fixed in #942 :-)
